### DISCUSSION TITLE
Communication Via Midi

### DISCUSSION THREAD

**Discussion by drjswalker** [FrancescoCeruti/linux-show-player#282] [Q&A]
G'day All,

I'm a magician and have been using LiSP for a few years as a Cue player but would like to try something a little harder in the form of tying a sound effect and a lighting effect together. I would like to play the sound effect (about 1-2s long) and at the same time have a light flash. I would like to use LiSP for the sound effect and have it send a MIDI cue/message to QLC+ to flash the light. I have each responding to a MIDI controller just fine. The part I'm having trouble with, is when I remove the controller from the equation and try to connect the MIDI message cue/message  to qlc+. I get an output that looks like "qlcplus:__QLC__ 128:0" in the MIDI devices output config box. But I don't see anything for LiSP in the MIDI segment of the INS/OUTS panel, Just two  entries labeled inputs(?) and an entry labeled "Midi through port-0". When it comes to settting the button widget properties in  the virtual desk, the autodetect sees nothing.

I am Setting the MIDI communication in Universe 4 and  DMX out in Universe 1, I am running the QLC+ virtual console when I test. I am setting the message parameters in the MIDI cue/message (Channel 13, note 6, Note on).

I am a midi very newbie but I have managed to get both Applications to respond independently to my Nanokontrol2 and do something useful. I think the problem is configuring the MIDI connection details between the two but I don't really know. I'll cross post this to a QLC group as well. I'm using Linux Mint Victoria and a recent version of Pipewire that I can confirm is working well.

Could someone please help connect these two so I can use the great MIDI messaging of LiSP to drive a QLC+ virtual console button and make an appearance on the stage truly magical

**Comment by FrancescoCeruti** [comment]
Hi! 
You should be able to see "RtMidi output" or something similar in QLC+, but you might need to refresh the inputs/outputs, or start QLC after you've opened LiSP.

If you set QLC as an output in LiSP and the "RtMidi output" as input in QLC and that should make the trick 🪄. I had some problems with note on/off, but you can use something else, like "program change".

You might need to redo the connection every time you open the apps.

Let me know :)
